src/header-generator.js

```diff
@@ -0,0 +1,313 @@
+const fs = require('fs');
+const path = require('path');
+const { BayesianNetwork } = require('bayesian-network');
+const { default: ow } = require('ow');
+
+const headerNetworkDefinitionPath = path.join(__dirname, './data_files/header-network-definition.json');
+const inputNetworkDefinitionPath = path.join(__dirname, './data_files/input-network-definition.json');
+const browserHelperFilePath = path.join(__dirname, './data_files/browser-helper-file.json');
+const headersOrderFilePath = path.join(__dirname, './data_files/headers-order.json');
+
+const browserHttpNodeName = '*BROWSER_HTTP';
+const operatingSystemNodeName = '*OPERATING_SYSTEM';
+const deviceNodeName = '*DEVICE';
+const missingValueDatasetToken = '*MISSING_VALUE*';
+
+const http2SecFetchAttributes = {
+    mode: 'sec-fetch-mode',
+    dest: 'sec-fetch-dest',
+    site: 'sec-fetch-site',
+    user: 'sec-fetch-user',
+};
+
+const http1SecFetchAttributes = {
+    mode: 'Sec-Fetch-Mode',
+    dest: 'Sec-Fetch-Dest',
+    site: 'Sec-Fetch-Site',
+    user: 'Sec-Fetch-User',
+};
+
+/*
+ * @private
+ */
+function getRandomInteger(minimum, maximum) {
+    return minimum + Math.floor(Math.random() * (maximum - minimum + 1));
+}
+
+/*
+ * @private
+ */
+function shuffleArray(array) {
+    if (array.length > 1) {
+        for (let x = 0; x < array.length; x++) {
+            const position1 = getRandomInteger(0, array.length - 1);
+            const position2 = getRandomInteger(0, array.length - 1);
+            const holder = array[position1];
+            array[position1] = array[position2];
+            array[position2] = holder;
+        }
+    }
+
+    return array;
+}
```

Comment:
I actually have the BayesianNetwork repository with absolute majority of the documentation done, I just needed to finish one thing before pushing it and creating a pull request for it too, but had my attention redirected elsewhere in the meantime. I had no idea someone would go through that code in this pull request, sorry :D 

I know the shuffling algorithm isn't great. It's nice that someone took the time to calculate how bad it is :D 
Anyway, it doesn't need to be great, I just use it to shuffle the locales a little so that they differ between various calls to getHeaders and I didn't want to add more dependencies needlessly.

As to whether I would rather use Python - I was actually toying with the idea of using it and making the generator a standalone script which would ran alongside the actor and communicate with him through message passing, but decided that this way it would be easier to use and that I don't actually need most of the advantages Python provides for it. Moreover, the repository that generates the definition files uses [DanfoJS](https://danfo.jsdata.org/), which is basically a JS copy of Pandas. The only problem is that it is still a little unfinished, so some things need to be worked around, but it was a great help anyway.

Reply:
Cool, I'll take a look at the commented BayesianNetwork version when it lands :) I was reading it in order to understand the logic of what's happening here, how does the generating work and what does it return and such.

As for the shuffling algorithm, I know it's not necessary to have it perfect in this case, there will usually probably be <10 locales in one array anyway, I'm just totally sure that one day in the future, when expanding this project, someone will need an array shuffling function for something else, will see that one is already implemented here, and will just use it without thinking twice. Anyway, how about something like this, adapted from Durstenfeld's array shuffling algorithm? It was included in Donald Knuth's The Art of Computer Programming, so I think it's reasonably good.

```javascript
function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
    }
}
```

Reply:
@fnesveda You actually read Donald Knuth's Art of Programming? 🙇

Reply:
@mnmkng Nah, I'm not that hardcore, but it was referenced to us a lot at school as "the algorithms bible" by many professors, and it kinda stuck, so I kinda implicitely trust stuff that comes from there :smile:

Reply:
@fnesveda Sure, will definitely request your review on that one :)
Good point with others wanting to use that function not knowing it is not the best possible algorithm. The algorithm you suggested looks good, so I used it.